packages/personal-data-erasure/src/main.js

```diff
@@ -106,92 +134,111 @@ export default class PersonalDataErasure {
         results = [...messages, ...results]
       }
       this.logger.info('Export operation completed successfully')
+
       return Promise.resolve(results)
     })
   }
 
-  async _getAllMessages(request) {
-    const messages = await this.client.process(request, response => {
-      if (response.statusCode !== 200 && response.statusCode !== 404)
-        return Promise.reject(
-          Error(`Request returned status code ${response.statusCode}`)
-        )
+  async _getAllMessages(request: ClientRequest): Promise<Messages> {
+    const messages = await this.client.process(
+      request,
+      (response: ClientResult): Promise<any> => {
+        if (response.statusCode !== 200 && response.statusCode !== 404)
+          return Promise.reject(
+            Error(`Request returned status code ${response.statusCode}`)
```

Comment:
hmm.. I could've caught this earlier.
Doesn't the API return a `message` as well? doesn't it make sense to display that together with the `statusCode`